src/SonarLint.VSSpecificAssemblies.props

```diff
@@ -15,7 +16,60 @@
             This is so the assemblies get copied locally when building the test projects so the tests can find them when they execute.
 
         3. Import this file
-         -->
+
+
+       Building from the command line
+       ==============================
+       When building from the command line you must specify the version of VS to target e.g.
+
+         msbuild.exe SonarLint.VisualStudio.Integration.sln /p:VsTargetVersion=2017   
+
+       The allowed values for VsTargetVersion are 2015, 2017 and 2019.
+
+
+       Command line build limitations
+       ==============================
+       MSBuild 14 must be used to build the VS2015 vsix, and it can only build that version.
+       MSBuild 15 and 16 can be used to build both VS2017 and VS2019 versions.
+       This limitation is due to a change the vsix manifest format after VS2015.
+ 
+
+       Building inside VS
+       ==================
+       The targets will automatically set the target VS version to the version of VS being used.
+
+  -->
+  
+  <PropertyGroup>
+    <!-- VsTargetVersion should be set when building from the command line to specify the version.
+         Expected value: 2015, 2017, 2019 -->
+    <VersionSpecificSuffix>$(VsTargetVersion)</VersionSpecificSuffix>
+
+    <VersionSpecificSuffix Condition="$(VersionSpecificSuffix)=='' AND $(VisualStudioVersion)=='14.0'">2015</VersionSpecificSuffix>
+    <VersionSpecificSuffix Condition="$(VersionSpecificSuffix)=='' AND $(VisualStudioVersion)=='15.0'">2017</VersionSpecificSuffix>
+    <VersionSpecificSuffix Condition="$(VersionSpecificSuffix)=='' AND $(VisualStudioVersion)=='16.0'">2019</VersionSpecificSuffix>
+  </PropertyGroup>
+
+  <!-- Sanity check to fail the build if VsTargetVersion isn't used correctly -->
+  <Target Name="CheckVsTargetVersionSetting" BeforeTargets="PrepareForBuild">
+    <Error 
+      Condition=" $(BuildingInsideVisualStudio) != 'true' AND $(VsTargetVersion) == '' "
+      Text="Must set VsTargetVersion build property when building outside Visual Studio. Expected values: 2015, 2017, 2019"/>
+
+    <Error
+      Condition=" $(BuildingInsideVisualStudio) != 'true' AND $(VsTargetVersion) != '2015' AND $(VsTargetVersion) != '2017' AND $(VsTargetVersion) != '2019' "
+      Text="Invalid value for VsTargetVersion. Expected values: 2015, 2017, 2019. Actual value: $(VsTargetVersion)"/>
+    
+    <!-- Check build constraints -->
```

Comment:
I added a couple more checks to make it easier to diagnose build errors due to trying to build invalid combinations e.g. MSBuild 14 + VS2019.